log/audit-logger.go

```diff
@@ -96,90 +113,118 @@ func GetAuditLogger() *AuditLogger {
 	return _Singleton.log
 }
 
-// Audit sends a NOTICE-severity message that is prefixed with the
-// audit tag, for special handling at the upstream system logger.
-func (log *AuditLogger) Audit(msg string) (err error) {
-	fmt.Println(msg)
-	err = log.Notice(fmt.Sprintf("%s %s", auditTag, msg))
-
-	log.Stats.Inc("Logging.Audit", 1, 1.0)
-
+// Log the provided message at the appropriate level, writing to
+// both stdout and the Logger, as well as informing statsd.
+func (log *AuditLogger) logAtLevel(level, msg string) (err error) {
+	fmt.Printf("%s\n", msg)
+	log.Stats.Inc(level, 1, 1.0)
+
+	switch level {
+	case "Logging.Alert":
+		err = log.Writer.Alert(msg)
+	case "Logging.Crit":
+		err = log.Writer.Crit(msg)
+	case "Logging.Debug":
+		err = log.Writer.Debug(msg)
+	case "Logging.Emerg":
+		err = log.Writer.Emerg(msg)
+	case "Logging.Err":
+		err = log.Writer.Err(msg)
+	case "Logging.Info":
+		err = log.Writer.Info(msg)
+	case "Logging.Warning":
+		err = log.Writer.Warning(msg)
+	}
 	return
 }
 
-// Audit can format an error for auditing; it does so at ERR level.
-func (log *AuditLogger) AuditErr(msg error) (err error) {
-	fmt.Println(msg)
-	err = log.Err(fmt.Sprintf("%s %s", auditTag, msg))
-
+// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
+func (log *AuditLogger) auditAtLevel(level, msg string) (err error) {
+	// Submit a separate counter that marks an Audit event
 	log.Stats.Inc("Logging.Audit", 1, 1.0)
 
-	return
+	text := fmt.Sprintf("%s %s", auditTag, msg)
+	return log.logAtLevel(level, text)
 }
 
-// Warning formats an error for the Warn level.
-func (log *AuditLogger) WarningErr(msg error) (err error) {
-	fmt.Println(msg)
-	err = log.Warning(fmt.Sprintf("%s", msg))
+// AuditPanic catches panics executables. This method should be added
+// in a defer statement as early as possible
+// AUDIT[ Error Conditions ] 9cc4d537-8534-4970-8665-4b382abe82f3
+func (log *AuditLogger) AuditPanic() {
+	if err := recover(); err != nil {
+		log.Audit(fmt.Sprintf("Panic: %v", err))
+	}
+}
 
-	return
+// WarningErr formats an error for the Warn level.
+func (log *AuditLogger) WarningErr(msg error) (err error) {
+	return log.logAtLevel("Logging.Warning", fmt.Sprintf("%s", msg))
```

Comment:
Oh, nevermind: I see this is about coercing an `error` into a string. Looks good.